Sources/BraintreePayPal/BTPayPalClient.swift

```diff
@@ -236,7 +240,17 @@ import BraintreeDataCollector
     // MARK: - App Switch Methods
 
     func handleReturnURL(_ url: URL) {
-        // TODO: implement handling return URL in a follow up PR
+        guard let returnURL = BTPayPalAppSwitchReturnURL(url: url) else {
+            notifyFailure(with: BTPayPalError.invalidURL("App Switch return URL cannot be nil"), completion: appSwitchCompletion)
+            return
+        }
+
+        switch returnURL.state {
+        case .succeeded, .canceled:
+            handleReturn(url, paymentType: .vault, completion: appSwitchCompletion)
+        default:
```

Comment:
Nit - more readable with full case name vs default
```suggestion
        case .unknown:
```

Reply:
Addressed here: https://github.com/braintree/braintree_ios/pull/1237/commits/bcbe0eb81fcfeda45271a53c6e2b26ea2805b490